src/lib/utils/styling/styler.ts

```diff
@@ -0,0 +1,128 @@
+   *
+   * Check inline style first then check computed (stylesheet) style
+   */
+  getFlowDirection(target: HTMLElement): [string, string] {
+    const query = 'flex-direction';
+    let value = this.lookupStyle(target, query);
+    if (value === FALLBACK_STYLE) {
+      value = '';
+    }
+    const hasInlineValue = this.lookupInlineStyle(target, query) ||
+    (isPlatformServer(this._platformId) && this._serverModuleLoaded) ? value : '';
+
+    return [value || 'row', hasInlineValue];
+  }
+
+  /**
+   * Find the DOM element's raw attribute value (if any)
+   */
+  lookupAttributeValue(element: HTMLElement, attribute: string): string {
+    return element.getAttribute(attribute) || '';
+  }
+
+  /**
+   * Find the DOM element's inline style value (if any)
+   */
+  lookupInlineStyle(element: HTMLElement, styleName: string): string {
+    return element.style[styleName] || element.style.getPropertyValue(styleName);
+  }
+
+  /**
+   * Determine the inline or inherited CSS style
+   * @TODO(CaerusKaru): platform-server has no implementation for getComputedStyle
+   */
+  lookupStyle(element: HTMLElement, styleName: string, inlineOnly = false): string {
+    let value = '';
+    if (element) {
+      let immediateValue = value = this.lookupInlineStyle(element, styleName);
+      if (!immediateValue) {
+        if (isPlatformBrowser(this._platformId)) {
+          if (!inlineOnly) {
+            value = getComputedStyle(element).getPropertyValue(styleName);
+          }
+        } else {
+          if (this._serverModuleLoaded) {
+            value = `${this._serverStylesheet.getStyleForElement(element, styleName)}`;
+          }
+        }
+      }
+    }
+
+    // Note: 'inline' is the default of all elements, unless UA stylesheet overrides;
+    //       in which case getComputedStyle() should determine a valid value.
+    return value ? value.trim() : FALLBACK_STYLE;
+  }
+
+  /**
+   * Applies the styles to the element. The styles object map may contain an array of values
+   * Each value will be added as element style
+   * Keys are sorted to add prefixed styles (like -webkit-x) first, before the standard ones
+   */
+  private _applyMultiValueStyleToElement(styles: {}, element: HTMLElement) {
+    Object.keys(styles).sort().forEach(key => {
+      const values = Array.isArray(styles[key]) ? styles[key] : [styles[key]];
+      values.sort();
+      for (let value of values) {
+        if (isPlatformBrowser(this._platformId) || !this._serverModuleLoaded) {
+          element.style.setProperty(key, value);
+        } else {
+          this._serverStylesheet.addStyleToElement(element, key, value);
+        }
+      }
+    });
+  }
+}
+
+/**
+ * Definition of a css style. Either a property name (e.g. "flex-basis") or an object
+ * map of property name and value (e.g. {display: 'none', flex-order: 5})
+ */
+export type StyleDefinition = string | { [property: string]: string | number | null };
```

Comment:
Why do either an object literal or string instead of just always doing an object literal?

Reply:
This is legacy code, removed. (The idea was that when calling a function, you could call it with string and value individually instead of providing an object)